typescript/cli/src/utils/keys.ts

```diff
@@ -14,34 +12,32 @@ const IMPERSONATED_KEY_TYPE = 'address';
  * Retrieves a signer for the current command-context.
  * @returns the signer
  */
-export async function getSigner<P extends ContextSettings>({
-  keyConfig,
+export async function getSigner({
+  key,
   skipConfirmation,
-}: P): Promise<providers.JsonRpcSigner | ethers.Wallet | undefined> {
-  if (!keyConfig) return undefined;
-
-  const key = await retrieveKey(DEFAULT_KEY_TYPE, keyConfig, skipConfirmation);
-
-  return privateKeyToSigner(key);
+}: {
+  key?: string;
+  skipConfirmation?: boolean;
+}) {
+  key ||= await retrieveKey(DEFAULT_KEY_TYPE, skipConfirmation);
+  const signer = privateKeyToSigner(key);
+  return { key, signer };
 }
 
 /**
  * Retrieves an impersonated signer for the current command-context.
  * @returns the impersonated signer
  */
-export async function getImpersonatedSigner<P extends ContextSettings>({
-  keyConfig,
+export async function getImpersonatedSigner({
+  key,
   skipConfirmation,
-}: P): Promise<providers.JsonRpcSigner | ethers.Wallet | undefined> {
-  if (!keyConfig) return undefined;
-
-  const key = await retrieveKey(
-    IMPERSONATED_KEY_TYPE,
-    keyConfig,
-    skipConfirmation,
-  );
-
-  return await addressToImpersonatedSigner(key);
+}: {
+  key?: string;
+  skipConfirmation?: boolean;
+}) {
+  key ||= await retrieveKey(IMPERSONATED_KEY_TYPE, skipConfirmation);
+  const signer = await addressToImpersonatedSigner(key);
```

Comment:
is key address here? I know this isnt your code but it looks confusing

Reply:
It is a bit confusing. Yes, the key param was overloaded to actually be an address for dry runs. 
We could likely improve this but I'm going to leave things as they are for now